support/tests/test_tealium_moments/MomentsAPITests.swift

```diff
@@ -0,0 +1,184 @@
+    }
+    
+    func testFetchEngineResponseForbidden() {
+        let session = MockURLSession()
+        let api = TealiumMomentsAPI(region: .us_east, account: MomentsAPITests.account, profile: MomentsAPITests.profile, environment: MomentsAPITests.environment, session: session)
+        let jsonData = "".data(using: .utf8)!
+        session.data = jsonData
+        session.response = HTTPURLResponse(url: URL(string: "https://example.com")!, statusCode: 403, httpVersion: nil, headerFields: nil)
+        session.error = nil
+        
+        let expectation = XCTestExpectation(description: "Fetch engine response fails")
+        
+        api.visitorId = "12345"
+        api.fetchEngineResponse(engineID: "engine123") { result in
+            switch result {
+            case .success:
+                XCTFail("Expected error, got success.")
+            case .failure(let error):
+                XCTAssertEqual((error as! MomentsAPIHTTPError), MomentsAPIHTTPError.forbidden, "Unexpected error type returned.")
+                expectation.fulfill()
+            }
+        }
+        
+        wait(for: [expectation], timeout: 1.0)
+    }
+    
+    func testFetchEngineResponseBadRequest() {
+        let session = MockURLSession()
+        let api = TealiumMomentsAPI(region: .us_east, account: MomentsAPITests.account, profile: MomentsAPITests.profile, environment: MomentsAPITests.environment, session: session)
+        let jsonData = "".data(using: .utf8)!
+        session.data = jsonData
+        session.response = HTTPURLResponse(url: URL(string: "https://example.com")!, statusCode: 400, httpVersion: nil, headerFields: nil)
+        session.error = nil
+        
+        let expectation = XCTestExpectation(description: "Fetch engine response fails")
+        
+        api.visitorId = "12345"
+        api.fetchEngineResponse(engineID: "engine123") { result in
+            switch result {
+            case .success:
+                XCTFail("Expected error, got success.")
+            case .failure(let error):
+                XCTAssertEqual((error as! MomentsAPIHTTPError), MomentsAPIHTTPError.badRequest, "Unexpected error type returned.")
+                expectation.fulfill()
+            }
+        }
+        
+        wait(for: [expectation], timeout: 1.0)
+    }
+    
+    func testFetchEngineResponseCustomReferer() {
+        let session = MockURLSessionCustomReferer()
+        let api = TealiumMomentsAPI(region: .us_east, account: MomentsAPITests.account, profile: MomentsAPITests.profile, environment: MomentsAPITests.environment, referer: "https://tealium.com/", session: session)
+        let jsonData = "".data(using: .utf8)!
+        session.data = jsonData
+        session.response = HTTPURLResponse(url: URL(string: "https://example.com")!, statusCode: 400, httpVersion: nil, headerFields: nil)
+        session.error = nil
+        
+        let expectation = XCTestExpectation(description: "Custom referer is correctly set")
```

Comment:
How are we checking that the custom referrer is actually set in the request?

Reply:
There's an assertion in the MockURLSessionCustomReferer.